### DISCUSSION TITLE
[BUG] Connection failure when LAN of server changes

### DISCUSSION THREAD

**Discussion by watchingdogs** [pivpn/pivpn#1422] [Feature requests]
## In raising this issue, I confirm the following: 

- [x] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting *is* directly related to the pivpn installer script.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

**Describe the bug**
The file `/etc/pivpn/wireguard/setupVars.conf` contains the hard-coded value of the router and the IPv4 address of the server, which causes connection issues. The issue was fixed on my end, by updating the values and restarting the server.
An update to the debug script that checks if the values in the file are still valid, and if not it updates them would solve the issue.

**To Reproduce**
Buy a new router. 

**Expected behavior**
Connection issues arise.

**Additional context**

```
$ uname -a
Linux raspberrypi 5.10.63-v7l+ #1459 SMP Wed Oct 6 16:41:57 BST 2021 armv7l GNU/Linux
```

**Comment by coolapso** [comment]
sorry but i can't see how this is a bug! at best this is a feature request and should be dealt like such here: 

https://github.com/pivpn/pivpn/discussions/categories/feature-requests

also at best we could warn users of such difference but we should not update them, that's up to you if you want to Re-configure your router with same network settings as before or if you want to update the `setupVars.conf` if you replace your router you always have to re-configure port forwarding anyways and we definitely won't touch on anything router related! 

Moved to feature requests.

**Reply by coolapso** [reply]
after giving it a 2 more minutes of thought ... I don't think this should even be implemented as there are too many things out of our control that can trigger false warnings or no warnings at all ... 

When you are setting up a server you have 2 typical scenarios: 

1 - Static IP configured directly on the server
2 - Using DHCP Reservation

when 2 applies its fine as the new router will give a different IP to the server and we could diff them however, 

When 1 applies its not possible to verify the the change in the router as the server will always be using the same IP Configurations regardless of the change on the router and there is no way for us to detect if the router was changed or not!

On Top of this ... when you change the router and if this router is directly interfacing the ISP Network, ISPs tend to change the External IP address which means more things have to be changed on both VPN Server and Clientes configurations. 

Changing networking environment is a user **option** and all the necessary precautions and investigations should be done before and necessary actions should be taken after.
